etcdserver/server.go

```diff
@@ -351,6 +354,10 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 		cl.SetStore(st)
 		cl.SetBackend(be)
 		cl.Recover()
+		if cl.Version != nil && cl.Version().LessThan(semver.Version{Major: 3}) && !beExist {
+			os.RemoveAll(bepath)
```

Comment:
not for cleanup. it is for checking. if the cluster version is 3, the be must exist before.

for the racey concern, this is a just a safety guard. We will check the consistent index of the beckend anyway. so even if we fail to remove it, the new backend will not contain the correct consistent index.